goss.yaml

```diff
@@ -0,0 +1,16 @@
+command:
+  self-tests-health:
+    exec: juju ssh loki-k8s/0 /charm/bin/pebble health
+    exit-status: 0
+    stdout:
+      - healthy
+  relation-cluster:
+    exec: juju status --relations | grep grafana_datasource | grep regular
+    exit-status: 0
+    stdout:
+      - grafana-source
+  charm-exit-codes:
+    exec: kubectl describe -n welcome-k8s pod loki-0 | grep -A 15 "Containers:" | grep -A 15 "charm:" | grep -A 3 "State:"
+    exit-status: 0
+    stdout:
+      - Running
```

Comment:
Does goss have json/yaml checks? E.g. we exec `kubectl -o json` and then the assertion is one something like '.containers.charm.state'.

Reply:
Yes goss has the following output options when running `validate`:

(default: "rspecish")
* documentation
* json
* junit
* nagios
* prometheus
* rspecish
* silent
* structured
* tap

However, in the case that we just exec `kubectl -o json` we should probably require the operator's host to have JQ and YQ installed because we will likely have a lot of exec commands like `kubectl -o json | jq -r '.containers.charm.state'` and the stdout check will have to reflect the expected value (`active/idle` in the example above).